aws/elasticsearch_domain_structure.go

```diff
@@ -0,0 +1,55 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	elasticsearch "github.com/aws/aws-sdk-go/service/elasticsearchservice"
+)
+
+func expandAdvancedSecurityOptions(m []interface{}) *elasticsearch.AdvancedSecurityOptionsInput {
+	config := elasticsearch.AdvancedSecurityOptionsInput{}
+	group := m[0].(map[string]interface{})
+
+	if v, ok := group["enabled"].(bool); ok {
+		config.Enabled = aws.Bool(v)
+	}
+
+	if v, ok := group["internal_user_database_enabled"].(bool); ok {
+		config.InternalUserDatabaseEnabled = aws.Bool(v)
+	}
+
+	if v, ok := group["master_user_options"].([]interface{}); ok {
+		if len(v) > 0 && v[0] != nil {
+			muo := elasticsearch.MasterUserOptions{}
+			masterUserOptions := v[0].(map[string]interface{})
+
+			if v, ok := masterUserOptions["master_user_arn"].(string); ok && v != "" {
+				muo.MasterUserARN = aws.String(v)
+			}
+
+			if v, ok := masterUserOptions["master_user_name"].(string); ok && v != "" {
+				muo.MasterUserName = aws.String(v)
+			}
+
+			if v, ok := masterUserOptions["master_user_password"].(string); ok && v != "" {
+				muo.MasterUserPassword = aws.String(v)
+			}
+
+			config.SetMasterUserOptions(&muo)
+		}
+	}
```

Comment:
let's also move this this logic w/in an `if` block, so we only set these fields iff  `enabled` or `v` in this case is true